src/graph/unit_graph.cc

```diff
@@ -83,6 +83,7 @@ class UnitGraph::COO : public BaseHeteroGraph {
     : BaseHeteroGraph(metagraph), adj_(coo) {
     // Data index should not be inherited. Edges in COO format are always
     // assigned ids from 0 to num_edges - 1.
+    CHECK(!COOHasData(coo)) << "[BUG] COO should not contain data.";
```

Comment:
Hmm... have you tried creating a sampled subgraph by COO?

Reply:
I rewrote the subgraph construction here:
https://github.com/dmlc/dgl/blob/022ddb65464726125ac861f58d38f1fee29bfe42/src/graph/sampling/neighbor/neighbor.cc#L82-L84